katas/es6/language/array-api/fill.js

```diff
@@ -35,4 +35,22 @@ describe('`Array.prototype.fill` can fill up an array with one value', () => {
     assert.deepEqual(arr, [1, 42, 3]);
   });
 
+
+  it('is a mutable method, and will change this object itself', function() {
+    const arr = new Array(3);
+    //// arr.fill('foo')
+    arr.fill(8);
+
+    assert.deepEqual(arr, [8,8,8]);
+  });
+  
+  it('when gets passed an object, it will copy the reference and fill the array with references to that object.', function(){
+    const arr = Array(3).fill({});
+    //// arr[0].foo = 'bar'
+    arr[0].hi = "es6 katas"; 
+    
+    assert.deepEqual(arr[2].hi, 'es6 katas');
+  });
```

Comment:
I don't understand the test description and how it is supposed to make the user arrive at the solution.
What I understand from the test description would be the following code (passing test).
```
const obj = {};
const arr = Array(1).fill(obj);
assert.equal(arr[0], obj);
```

wouldn't it?

Reply:
Yes, you're right 😂